taskcluster/utils.py

```diff
@@ -276,6 +317,23 @@ def authenticate(description=None):
     The description will be shown on login.taskcluster.net, if not provided
     a default message with script path will be displayed.
   """
+
+  # Let's first offer to read credentials from a configuration file then
+  # the environment
+  existingCred = {}
+  try:
+    existingCred.update(readCfgFile())
+  except:
+    log.warn('Exception thrown while trying to read credentials from file')
+
+  try:
+    existingCred.update(readCredentialsFromEnv())
+  except:
+    log.warn('Exception thrown while trying to read credentials from environment')
+
+  if 'clientId' in existingCred and 'accessToken' in existingCred:
+    return existingCred
```

Comment:
Oh, I almost forget to mention this... **We must check `isExpired`** if the credentials have a certificate.

Otherwise, we can return credentials that won't work, which isn't much fun...